apps/web/app/(app)/environments/[environmentId]/settings/(account)/profile/components/EditProfileDetailsForm.tsx

```diff
@@ -1,136 +1,209 @@
 
-export const EditProfileDetailsForm = ({ user }: { user: TUser }) => {
+export const EditProfileDetailsForm = ({
+  user,
+  emailVerificationDisabled,
+}: {
+  user: TUser;
+  emailVerificationDisabled: boolean;
+}) => {
+  const { t } = useTranslate();
+  const router = useRouter();
+
   const form = useForm<TEditProfileNameForm>({
-    defaultValues: { name: user.name, locale: user.locale || "en" },
+    defaultValues: {
+      name: user.name,
+      locale: user.locale || "en",
+      email: user.email,
+    },
     mode: "onChange",
     resolver: zodResolver(ZEditProfileNameFormSchema),
   });
 
   const { isSubmitting, isDirty } = form.formState;
-  const { t } = useTranslate();
 
-  const onSubmit: SubmitHandler<TEditProfileNameForm> = async (data) => {
+  const [showModal, setShowModal] = useState(false);
+  const [pendingData, setPendingData] = useState<TEditProfileNameForm | null>(null);
+
+  const handleConfirmPassword = async (password: string) => {
+    if (!pendingData) return;
+
     try {
-      const name = data.name.trim();
-      const locale = data.locale;
-      await updateUserAction({ name, locale });
-      toast.success(t("environments.settings.profile.profile_updated_successfully"));
+      if (pendingData.email !== user.email) {
+        const signInResponse = await signIn("credentials", {
+          email: user.email,
+          password,
+          redirect: false,
+        });
+
+        if (!signInResponse || signInResponse.error) {
+          toast.error("Invalid password");
+          return;
+        }
+
+        if (!emailVerificationDisabled) {
+          await updateUserAction({ name: pendingData.name, locale: pendingData.locale });
+          const response = await sendVerificationNewEmailAction({ email: pendingData.email });
+          if (response?.data) {
+            toast.success(t("auth.verification-requested.verification_email_successfully_sent"));
+          } else {
+            toast.error(getFormattedErrorMessage(response));
+          }
+        } else {
+          toast.success(t("environments.settings.profile.profile_updated_successfully"));
+          await signOut({ redirect: false });
+          const url = `/email-change-without-verification-success`;
+          router.push(url);
+          return;
+        }
+      } else {
+        await updateUserAction({ ...pendingData });
+        toast.success(t("environments.settings.profile.profile_updated_successfully"));
+      }
+
       window.location.reload();
-      form.reset({ name, locale });
-    } catch (error) {
+      setShowModal(false);
+    } catch (error: any) {
       toast.error(`${t("common.error")}: ${error.message}`);
     }
   };
 
+  const onSubmit: SubmitHandler<TEditProfileNameForm> = async (data) => {
+    if (data.email !== user.email) {
+      setPendingData(data);
+      setShowModal(true);
+    } else {
+      try {
+        await updateUserAction({ ...data });
```

Comment:
Please trim the `name` here before updating the user, if you're not doing it elsewhere.